app/assets/javascripts/core/user_form_controller.coffee

```diff
@@ -0,0 +1,27 @@
+#= require './view_controller'
+
+class Spaces.UserFormController extends Spaces.ViewController
+
+	initialize: ->
+		# Update the user form tags when a post is added
+		($ 'body').on 'postAdded', () ->
+			updateUserForm()
+		# Hide modal and update user form when it is submitted
+		($ 'body').on 'userUpdated', () ->
+			userFormSubmitted()
+		# If the form has a counter, update it
+		($ '#post_short').keyup (e) ->
+			($ '.counter').html ($ @).val().length 
+
+	userFormSubmitted = () ->
+		($ '.modal').modal 'hide'
+		updateUserForm()
```

Comment:
if you updated these as methods like I suggested you would use `@updateUserForm()`